### ISSUE TITLE
Pitfall of "Fill fields with most often used entries"

### ISSUE THREAD

**Issue by ReporterX** [stephanmahieu/formhistorycontrol-2#44]
It seems there are only two ways to fill **the whole form**. They are: 
1. Fill fields with most recent entries  
2. Fill fields with most often used entries  

There is a pitfall using the second one. Let's say you fill the form using the second one. The form filled is not what you expect. You want to try the first one (the most recent one). Unfortunately it is gone since the one you've just filled in (the most often used) becomes the most recent one now.

Perhaps a fix is it only counts as a new one when you start typing, not at the time of restoration.

Or list more than 1 most recent entries. Let's say the 5 most recent ones.  

Thank you.

**Comment by stephanmahieu** [comment]
Yes, I struggled with this issue. Say al you do is use option 2 and it is never counted. In time it looks likes some values have not been used for a long time and can even be cleanup. That is the reason why I chose to update the stats when values have been restored.
Using option 2 and discover it is not what you want indeed renders option 1 useless.

I like the idea of displaying some recent items in the context menu. I will add that in one of the upcoming releases.
